`glibc` version too damn high?

### Describe the bug

```
nu: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by nu)
nu: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by nu)
nu: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by nu)
```

The `glibc` version on current Debian 11 Bullseye (stable) is [2.31](https://packages.debian.org/bullseye/libc6).

Which means, the most recent version of `nu` cannot be installed through the officially recommended ways without unrecommended & potentially risky hacks to the system.

I've seen other issues like this before.

https://github.com/nushell/nushell/issues/3825
https://github.com/nushell/nushell/issues/3737

Usually, they could be fixed by upgrading the system's packages.
In this case, it does not seem possible, without issues.

### How to reproduce

1. Run Debian 11 Bullseye (stable, as of now).
2. `apt update && apt upgrade -y`
3. Still, `nu` won't run.

### Expected behavior

Continues to run on popular GNU/Linux distribution, even after an upgrade.

### Screenshots

_No response_

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.71.0                                          |
| branch             |                                                 |
| commit_hash        | 6cc4ef6c706ebbdc5b995013e0b00f77f7a4dba7        |
| build_os           | linux-x86_64                                    |
| build_target       | x86_64-unknown-linux-gnu                        |
| rust_version       | rustc 1.65.0 (897e37553 2022-11-02)             |
| rust_channel       | stable-x86_64-unknown-linux-gnu                 |
| cargo_version      | cargo 1.65.0 (4bc8f24d3 2022-10-20)             |
| pkg_version        | 0.71.0                                          |
| build_time         | 2022-11-08 18:00:26 +00:00                      |
| build_rust_channel | release                                         |
| features           | database, dataframe, default, trash, which, zip |
| installed_plugins  | partial                                         |

### Additional context

**Platform**
Debian GNU/Linux 11 (bullseye) in WSL2 on Windows 10

**Terminal software**
Windows Terminal

Nushell is running "natively" as a login shell.

____

I don't know, whether this is acceptable or not. Though, I had the impression that support for major operating systems in their currently most popular/recent edition is significant to this project's goals.

[setup-nu](https://github.com/hustcer/setup-nu) failed for nu v0.72: 
https://github.com/hustcer/setup-nu/actions/runs/3581284227/jobs/6024177325
```
Run $'Nu version info:(char nl)'; version
  $'Nu version info:(char nl)'; version
  shell: /opt/hostedtoolcache/nushell/0.7[2](https://github.com/hustcer/setup-nu/actions/runs/3581284227/jobs/6024177325#step:4:2).0/x64/nu {0}
/opt/hostedtoolcache/nushell/0.72.0/x64/nu: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.[3](https://github.com/hustcer/setup-nu/actions/runs/3581284227/jobs/6024177325#step:4:3)[4](https://github.com/hustcer/setup-nu/actions/runs/3581284227/jobs/6024177325#step:4:5)' not found (required by /opt/hostedtoolcache/nushell/0.72.0/x64/nu)
/opt/hostedtoolcache/nushell/0.72.0/x64/nu: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /opt/hostedtoolcache/nushell/0.72.0/x64/nu)
/opt/hostedtoolcache/nushell/0.72.0/x[6](https://github.com/hustcer/setup-nu/actions/runs/3581284227/jobs/6024177325#step:4:7)4/nu: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /opt/hostedtoolcache/nushell/0.[7](https://github.com/hustcer/setup-nu/actions/runs/3581284227/jobs/6024177325#step:4:8)2.0/x64/nu)
Error: Process completed with exit code 1.
```

--- a/.github/workflows/release-pkg.nu
+++ b/.github/workflows/release-pkg.nu
@@ -16,6 +16,9 @@ let flags = $env.TARGET_RUSTFLAGS
 let dist = $'($env.GITHUB_WORKSPACE)/output'
 let version = (open Cargo.toml | get package.version)
 
+$'Debugging info:'
+print { version: $version, bin: $bin, os: $os, target: $target, src: $src, flags: $flags, dist: $dist }; hr-line
+
 # $env
 
 $'(char nl)Packaging ($bin) v($version) for ($target) in ($src)...'; hr-line -b

--- a/.github/workflows/release-pkg.nu
+++ b/.github/workflows/release-pkg.nu
@@ -26,8 +26,8 @@ $'Start building ($bin)...'; hr-line
 # ----------------------------------------------------------------------------
 # Build for Ubuntu and macOS
 # ----------------------------------------------------------------------------
-if $os in ['ubuntu-latest', 'macos-latest'] {
-    if $os == 'ubuntu-latest' {
+if $os in ['ubuntu-22.04', 'macos-latest'] {
+    if $os == 'ubuntu-22.04' {
         sudo apt-get install libxcb-composite0-dev -y
     }
     if $target == 'aarch64-unknown-linux-gnu' {
@@ -41,7 +41,7 @@ if $os in ['ubuntu-latest', 'macos-latest'] {
     } else {
         # musl-tools to fix 'Failed to find tool. Is `musl-gcc` installed?'
         # Actually just for x86_64-unknown-linux-musl target
-        if $os == 'ubuntu-latest' { sudo apt install musl-tools -y }
+        if $os == 'ubuntu-22.04' { sudo apt install musl-tools -y }
         cargo-build-nu $flags
     }
 }
@@ -88,7 +88,7 @@ if ($ver | str trim | is-empty) {
 # Create a release archive and send it to output for the following steps
 # ----------------------------------------------------------------------------
 cd $dist; $'(char nl)Creating release archive...'; hr-line
-if $os in ['ubuntu-latest', 'macos-latest'] {
+if $os in ['ubuntu-22.04', 'macos-latest'] {
 
     let files = (ls | get name)
     let dest = $'($bin)-($version)-($target)'
@@ -101,7 +101,8 @@ if $os in ['ubuntu-latest', 'macos-latest'] {
 
     tar -czf $archive $dest
     print $'archive: ---> ($archive)'; ls $archive
-    echo $'::set-output name=archive::($archive)'
+    # REF: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
+    echo $"archive=($archive)" | save --append $env.GITHUB_OUTPUT
 
 } else if $os == 'windows-latest' {
 
@@ -121,7 +122,8 @@ if $os in ['ubuntu-latest', 'macos-latest'] {
         cp -r $'($dist)/*' target/release/
         cargo install cargo-wix --version 0.3.3
         cargo wix --no-build --nocapture --package nu --output $wixRelease
-        echo $'::set-output name=archive::($wixRelease)'
+        print $'archive: ---> ($wixRelease)';
+        echo $"archive=($wixRelease)" | save --append $env.GITHUB_OUTPUT
 
     } else {
 
@@ -131,7 +133,7 @@ if $os in ['ubuntu-latest', 'macos-latest'] {
         print $'archive: ---> ($archive)';
         let pkg = (ls -f $archive | get name)
         if not ($pkg | is-empty) {
-            echo $'::set-output name=archive::($pkg | get 0)'
+            echo $"archive=($pkg | get 0)" | save --append $env.GITHUB_OUTPUT
         }
     }
 }

--- a/.github/workflows/release-pkg.nu
+++ b/.github/workflows/release-pkg.nu
@@ -17,9 +17,11 @@ let dist = $'($env.GITHUB_WORKSPACE)/output'
 let version = (open Cargo.toml | get package.version)
 
 $'Debugging info:'
-print { version: $version, bin: $bin, os: $os, target: $target, src: $src, flags: $flags, dist: $dist }; hr-line
+print { version: $version, bin: $bin, os: $os, target: $target, src: $src, flags: $flags, dist: $dist }; hr-line -b
 
 # $env
+
+let USE_UBUNTU = 'ubuntu-20.04'
 
 $'(char nl)Packaging ($bin) v($version) for ($target) in ($src)...'; hr-line -b
 if not ('Cargo.lock' | path exists) { cargo generate-lockfile }
@@ -29,8 +31,8 @@ $'Start building ($bin)...'; hr-line
 # ----------------------------------------------------------------------------
 # Build for Ubuntu and macOS
 # ----------------------------------------------------------------------------
-if $os in ['ubuntu-22.04', 'macos-latest'] {
-    if $os == 'ubuntu-22.04' {
+if $os in [$USE_UBUNTU, 'macos-latest'] {
+    if $os == $USE_UBUNTU {
         sudo apt-get install libxcb-composite0-dev -y
     }
     if $target == 'aarch64-unknown-linux-gnu' {
@@ -44,7 +46,7 @@ if $os in ['ubuntu-22.04', 'macos-latest'] {
     } else {
         # musl-tools to fix 'Failed to find tool. Is `musl-gcc` installed?'
         # Actually just for x86_64-unknown-linux-musl target
-        if $os == 'ubuntu-22.04' { sudo apt install musl-tools -y }
+        if $os == $USE_UBUNTU { sudo apt install musl-tools -y }
         cargo-build-nu $flags
     }
 }
@@ -91,7 +93,7 @@ if ($ver | str trim | is-empty) {
 # Create a release archive and send it to output for the following steps
 # ----------------------------------------------------------------------------
 cd $dist; $'(char nl)Creating release archive...'; hr-line
-if $os in ['ubuntu-22.04', 'macos-latest'] {
+if $os in [$USE_UBUNTU, 'macos-latest'] {
 
     let files = (ls | get name)
     let dest = $'($bin)-($version)-($target)'

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -44,22 +44,22 @@ jobs:
           os: windows-latest
           target_rustflags: ''
         - target: x86_64-unknown-linux-gnu
-          os: ubuntu-latest
+          os: ubuntu-22.04
           target_rustflags: ''
         - target: x86_64-unknown-linux-musl
-          os: ubuntu-latest
+          os: ubuntu-22.04
           target_rustflags: ''
         - target: aarch64-unknown-linux-gnu
-          os: ubuntu-latest
+          os: ubuntu-22.04
           target_rustflags: ''
         - target: armv7-unknown-linux-gnueabihf
-          os: ubuntu-latest
+          os: ubuntu-22.04
           target_rustflags: ''
 
     runs-on: ${{matrix.os}}
 
     steps:
-    - uses: actions/checkout@v3.0.2
+    - uses: actions/checkout@v3.1.0
 
     - name: Install Rust Toolchain Components
       uses: actions-rs/toolchain@v1.0.6
@@ -70,9 +70,9 @@ jobs:
         target: ${{ matrix.target }}
 
     - name: Setup Nushell
-      uses: hustcer/setup-nu@v2.1
+      uses: hustcer/setup-nu@v3
       with:
-        version: 0.69.1
+        version: 0.72.0
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -44,16 +44,16 @@ jobs:
           os: windows-latest
           target_rustflags: ''
         - target: x86_64-unknown-linux-gnu
-          os: ubuntu-22.04
+          os: ubuntu-20.04
           target_rustflags: ''
         - target: x86_64-unknown-linux-musl
-          os: ubuntu-22.04
+          os: ubuntu-20.04
           target_rustflags: ''
         - target: aarch64-unknown-linux-gnu
-          os: ubuntu-22.04
+          os: ubuntu-20.04
           target_rustflags: ''
         - target: armv7-unknown-linux-gnueabihf
-          os: ubuntu-22.04
+          os: ubuntu-20.04
           target_rustflags: ''
 
     runs-on: ${{matrix.os}}
@@ -72,7 +72,7 @@ jobs:
     - name: Setup Nushell
       uses: hustcer/setup-nu@v3
       with:
-        version: 0.72.0
+        version: 0.71.0
       env:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 

